playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

```diff
@@ -671,6 +672,22 @@ private String getDecoderInitializationErrorMessage(ExoPlaybackException error,
         return errorMessage;
     }
 
+    private String getUnexpectedErrorMessage(ExoPlaybackException error, String errorMessage) {
+        Exception cause = error.getUnexpectedException();
+        if (cause.getCause() != null) {
+            errorMessage = cause.getCause().getMessage();
+        }
+        return errorMessage;
+    }
+
+    private String getSourceErrorMessage(ExoPlaybackException error, String errorMessage) {
+        Exception cause = error.getSourceException();
+        if (cause.getCause() != null) {
```

Comment:
also here and if error id null do not over ride the error with null plz

Reply:
We have this check in Line: 643